app/components/au-address-search.js

```diff
@@ -0,0 +1,383 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { task, timeout } from 'ember-concurrency';
+
+/**
+ * @typedef {Object} Address
+ * @property {string} country
+ * @property {string} province
+ * @property {string} municipality
+ * @property {string} postalCode
+ * @property {string} street
+ * @property {string} houseNumber
+ * @property {string | null} boxNumber
+ * @property {boolean | undefined } completed
+ */
+
+/**
+ * @typedef {Object} LocationInFlanders
+ * @property {string} municipality
+ * @property {string} postalCode
+ * @property {string} street
+ * @property {string} housenumber
+ */
+
+/**
+ * @typedef {'manual'|'automatic'} AddressSearchMode
+ */
+
+/** @type { (fieldName: keyof<Address>) => ((e:UIEvent)=>void) } */
+function handleSimpleManualInputEventChange(fieldName) {
+  return function (e) {
+    e.preventDefault();
+    const newValue = e.target.value ?? undefined;
+    this.selectedAddress = {
+      ...this.selectedAddress,
+      [fieldName]: newValue,
+    };
+  };
+}
+
+export default class AuAddressSearchComponent extends Component {
+  constructor(...args) {
+    super(...args);
+    // We use a task here because by the time ember concurrency invokes it the args will be
+    // passed to the component. When this constructor runs the ember framework has not yet
+    // passed `this.args`
+    this._initTask.perform();
+  }
+
+  /** Initialse the component with args */
+  _initTask = task(async () => {
+    if (this.addressOk) {
+      // If the passed address is already OK we pre fill the addrress suggestion
+      this.selectedAddressSuggestion = this.args.address;
+    }
+  });
+
+  get classes() {
+    if (this.args.errorMessage) return 'address-search-error-container';
+    if (this.args.warningMessage) return 'address-search-warning-container';
+    return '';
+  }
+
+  get extraMessage() {
+    if (this.args.errorMessage) return this.args.errorMessage;
+    if (this.args.warningMessage) return this.args.warningMessage;
+    return '';
+  }
+
+  /** @type { Partial<Address> } */
+  get selectedAddress() {
+    return this.args.address;
+  }
+
+  set selectedAddress(value) {
+    value.completed = this.addressOk;
+    this.args.onChange(value);
+  }
+
+  /**
+   * A little helper function to pretty print an address
+   * @param { Address } address
+   * @returns { string }
+   */
+  printAddress(address) {
+    return address.boxNumber
+      ? `${address.street} ${address.houseNumber} (Bus: ${address.boxNumber}), ${address.postalCode} ${address.municipality}`
+      : `${address.street} ${address.houseNumber}, ${address.postalCode} ${address.municipality}`;
+  }
+
+  /**
+   * Simple 'or' helper functuon
+   * @param { any } arg1
+   * @param { any } arg2
+   * @returns { boolean }
+   */
+  or(arg1, arg2) {
```

Comment:
consider adding this as a helper so we can use it in more places if needed, alternatively a lot of ember apps use this addon https://github.com/jmurphyau/ember-truth-helpers